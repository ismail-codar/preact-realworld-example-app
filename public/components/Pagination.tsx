import { articlePageLimit } from '../services/api/article';

interface PaginationProps {
	/** total items count */
	count: number;
	page: number;
	setPage: (page: number) => void;
}

export function Pagination(props: PaginationProps) {
	const pagesCount = Math.ceil(props.count / articlePageLimit);
	const countArray = new Array(pagesCount).fill('');
	const isActive = (index: number) => (props.page === index + 1 ? 'active' : '');

	return (
		<ul class="pagination">
			{countArray.map((_, index) => (
				<li key={index} class={`page-item ${isActive(index)}`}>
					<a onClick={() => props.setPage(index + 1)} class="page-link">
						{index + 1}
					</a>
				</li>
			))}
		</ul>
	);
}
