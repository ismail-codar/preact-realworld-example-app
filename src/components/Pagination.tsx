import { FunctionalComponent, h } from 'preact'
import { limit } from '../services'

interface PaginationProps {
  /** total items count */
  count: number;
  page: number;
  setPage: (page: number) => void;
}

const Pagination: FunctionalComponent<PaginationProps> = (props) => {
  const pagesCount = Math.ceil(props.count / limit)
  const countArray = new Array(pagesCount).fill('')
  const isActive = (index: number) => props.page === index + 1 ? 'active' : ''

  return (
    <ul className="pagination">
      {countArray.map((_, index) => (
        <li key={index} className={`page-item ${isActive(index)}`}>
          <a onClick={() => props.setPage(index + 1)} className="page-link">{index + 1}</a>
        </li>
      ))}
    </ul>
  )
}

export default Pagination
