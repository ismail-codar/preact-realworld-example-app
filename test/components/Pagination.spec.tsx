import { h } from 'preact';
import { fireEvent, render, screen } from '@testing-library/preact';

import Pagination from '../../src/components/Pagination';

const setPage = jest.fn();

test('renders the Pagination component', () => {
	const { asFragment } = render(<Pagination count={1} page={1} setPage={setPage} />);

	expect(asFragment()).toMatchSnapshot();
});

test('should display multiple items when count is passed in', () => {
	render(<Pagination count={50} page={1} setPage={setPage} />);

	expect(screen.getAllByRole('listitem')).toHaveLength(5);
});

test('should highlight current page that is passed in props', () => {
	render(<Pagination count={50} page={2} setPage={setPage} />);

	expect(screen.getAllByRole('listitem')[1]).toHaveClass('active');
});

test('should set current page when item clicked', () => {
	render(<Pagination count={50} page={1} setPage={setPage} />);

	fireEvent.click(screen.getByText('2'));

	expect(setPage).toBeCalledWith(2);
});
