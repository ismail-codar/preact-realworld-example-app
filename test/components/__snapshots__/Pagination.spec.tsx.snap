// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Pagination component 1`] = `
<DocumentFragment>
  <ul
    class="pagination"
  >
    <li
      class="page-item active"
    >
      <a
        class="page-link"
      >
        1
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
